skimage/future/graph/rag.py

```diff
@@ -51,6 +51,67 @@ def min_weight(graph, src, dst, n):
     return min(w1, w2)
 
 
+def mean_color(graph, extra_arguments=[], extra_keywords={}):
+    """Callback to compute edge weights.
+
+    The method expects that each node has already been described.
+
+    Parameters
+    ----------
+    graph : RAG
+        The graph under consideration.
+    extra_arguments : sequence, optional
+        Allows extra positional arguments passed.
+    extra_keywords : dictionary, optional
+        Allows extra keyword arguments passed.
+        sigma = 255
+        mode = 'distance'
+    """
+    sigma = extra_keywords['sigma']
+    mode = extra_keywords['mode']
+    for x, y, d in graph.edges_iter(data=True):
+        diff = graph.node[x]['mean color'] - graph.node[y]['mean color']
+        diff = np.linalg.norm(diff)
+        if mode == 'similarity':
+            d['weight'] = math.e ** (-(diff ** 2) / sigma)
+        elif mode == 'distance':
+            d['weight'] = diff
+        else:
+            raise ValueError("The mode '%s' is not recognised" % mode)
+
+
+def describe_node(graph,labels,image, *args, **kwargs):
+    """Callback to handle describing nodes.
+
+    Nodes can have arbitrary Python objects assigned as attributes. This method
+    expects a valid graph and computes the mean color of the node.
+
+    Parameters
+    ----------
+    graph : RAG
+        The graph under consideration.
+    extra_arguments : sequence, optional
+        Allows extra positional arguments passed.
+    extra_keywords : dictionary, optional
+        Allows extra keyword arguments passed.
+
+    """
+    for n in graph:
+        graph.node[n].update({'labels': [n],
+                              'pixel count': 0,
+                              'total color': np.array([0, 0, 0],
+                                                      dtype=np.double)})
+
+    for index in np.ndindex(labels.shape):
+        current = labels[index]
+        graph.node[current]['pixel count'] += 1
+        graph.node[current]['total color'] += image[index]
+
+    for n in graph:
+        graph.node[n]['mean color'] = (graph.node[n]['total color'] /
```

Comment:
Unless one of these values is a float, the `/` operator will be truncating division (i.e., `//`) under Python 2. They both look like ints to me, so that's probably not what you want - though I could be wrong.

If you do want truncating division, use `//`. If you want true division, then we should add `from __future__ import division` at the very top of this file which will normalize this behavior across Python major versions.